spec/lib/efile/nj/nj1040_calculator_spec.rb

```diff
@@ -623,12 +623,58 @@ def over_65_birth_year
     end
   end
 
+  describe 'line 28b - other retirement income exclusion' do
+    let(:intake) { create(:state_file_nj_intake) }
+
+    context 'when taxpayer is ineligible' do
+      it 'sets line 28b to 0' do
+        allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return false
+        instance.calculate
+        expect(instance.lines[:NJ1040_LINE_28B].value).to eq(0)
+      end
+    end
+
+    context 'when taxpayer is eligible' do
+      context 'when maximum_exclusion minus 28a is greater than total_eligible_nonretirement_income' do
+        it 'sets line 28b to total_eligible_nonretirement_income' do
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return true
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:total_eligible_nonretirement_income).and_return 1_000
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:calculate_maximum_exclusion).and_return 50_000
+          allow(instance).to receive(:calculate_line_28a).and_return 0
+          instance.calculate
+          expect(instance.lines[:NJ1040_LINE_28B].value).to eq(1000)
+        end
+      end
+
+      context 'when maximum_exclusion minus 28a is less than total_eligible_nonretirement_income' do
+        it 'sets line 28b to maximum_exclusion minus 28a' do
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return true
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:total_eligible_nonretirement_income).and_return 100_000
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:calculate_maximum_exclusion).and_return 50_000
+          allow(instance).to receive(:calculate_line_28a).and_return 0
```

Comment:
[pebble] Right now, by mocking 28a to 0, we don't actually receive any confidence from this test that 28a is incorporated into the calculation at all - I think we could remove it from the code and the test would still pass. If we want to confirm that all parts are working as expected, something like max_exclusion=50,000, 28a=1,000, total_eligible=49,500. This would then ONLY pass if 28a is included in the calculation